source/include/ice_data_types.h

```diff
@@ -0,0 +1,184 @@
+ */
+#define ICE_MAX_CONFIG_CREDENTIAL_LEN                              256
+
+/**
+ * Maximum allowed ICE URI length
+ */
+#define ICE_MAX_CONFIG_URI_LEN                                     256
+
+
+#define ICE_IS_IPV4_ADDR( pAddress )    ( ( pAddress ).family == STUN_ADDRESS_IPv4 )
+
+#define ICE_STUN_MESSAGE_BUFFER_SIZE    1024
+
+typedef uint64_t ( * Ice_ComputeRandom ) ( void );
+
+typedef uint32_t ( * Ice_ComputeCrc32 ) ( uint32_t initialResult,
+                                          uint8_t * pBuffer,
+                                          uint32_t bufferLength );
+typedef void ( * Ice_ComputeHMAC ) ( uint8_t * pPassword,
+                                     uint32_t passwordLength,
+                                     uint8_t * pBuffer,
+                                     uint32_t bufferLength,
+                                     uint8_t * pOutput,
+                                     uint32_t * pOutputLength );
+typedef enum
+{
+    ICE_CANDIDATE_TYPE_HOST,
+    ICE_CANDIDATE_TYPE_PEER_REFLEXIVE,
+    ICE_CANDIDATE_TYPE_SERVER_REFLEXIVE,
+    ICE_CANDIDATE_TYPE_RELAYED,
+} IceCandidateType_t;
+
+typedef enum
+{
+    ICE_CANDIDATE_STATE_INVALID,
+    ICE_CANDIDATE_STATE_NEW,
+    ICE_CANDIDATE_STATE_VALID
+} IceCandidateState_t;
+
+typedef enum
+{
+    ICE_CANDIDATE_PAIR_STATE_INVALID,
+    ICE_CANDIDATE_PAIR_STATE_FROZEN,
+    ICE_CANDIDATE_PAIR_STATE_WAITING,
+    ICE_CANDIDATE_PAIR_STATE_VALID,
+    ICE_CANDIDATE_PAIR_STATE_NOMINATED,
+    ICE_CANDIDATE_PAIR_STATE_SUCCEEDED
+} IceCandidatePairState_t;
+
+typedef enum
+{
+    ICE_SOCKET_PROTOCOL_NONE,
+    ICE_SOCKET_PROTOCOL_TCP,
+    ICE_SOCKET_PROTOCOL_UDP
+} IceSocketProtocol_t;
+
+typedef enum IceResult
+{
+    ICE_RESULT_OK,
+    ICE_RESULT_BAD_PARAM,
+    ICE_RESULT_MAX_CANDIDATE_THRESHOLD,
+    ICE_RESULT_MAX_CANDIDATE_PAIR_THRESHOLD,
+    ICE_RESULT_OUT_OF_MEMORY
+} IceResult_t;
+
+typedef enum IceStunPacketHandleResult
+{
+    ICE_RESULT_STUN_PACKET_ERROR,
+    ICE_RESULT_FOUND_PEER_REFLEXIVE_CANDIDATE,
+    ICE_RESULT_UPDATED_SRFLX_CANDIDATE_ADDRESS,
+    ICE_RESULT_SEND_RESPONSE_FOR_REMOTE_REQUEST,
+    ICE_RESULT_SEND_TRIGGERED_CHECK,
+    ICE_RESULT_START_NOMINATION,
+    ICE_RESULT_USE_CANDIDATE_FLAG,
+    ICE_RESULT_SEND_RESPONSE_FOR_NOMINATION,
+    ICE_RESULT_CANDIDATE_PAIR_READY,
+    ICE_RESULT_NOT_FOUND_CANDIDATE,
+    ICE_RESULT_NOT_FOUND_CANDIDATE_PAIR,
+    ICE_RESULT_NOT_FOUND_ADDRESS_ATTRIBUTE,
+    ICE_RESULT_NOT_FOUND_MATCHING_TRANSACTION_ID,
+    ICE_RESULT_STUN_BINDING_INDICATION
+} IceStunPacketHandleResult_t;
+
+/* ICE component structures */
+
+typedef struct IceIPAddress
+{
+    StunAttributeAddress_t ipAddress;
+    uint32_t isPointToPoint;
+} IceIPAddress_t;
+
+typedef struct TransactionIdStore
+{
+    uint32_t maxTransactionIdsCount;
+    uint32_t nextTransactionIdIndex;
+    uint32_t earliestTransactionIdIndex;
+    uint32_t transactionIdCount;
+    uint8_t * pTransactionIds;
+} TransactionIdStore_t;
```

Comment:
Are we still using transaction ID store in this implementation?

Reply:
Yes, we are keeping the Transaction Store for the STUN packets in the ICE agent